src/test/regress/expected/failure_ddl.out

```diff
@@ -0,0 +1,1228 @@
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100804,t,3)
+ (localhost,57637,100805,t,3)
+ (localhost,57637,100806,t,3)
+ (localhost,57637,100807,t,3)
+ (localhost,57640,100804,t,3)
+ (localhost,57640,100805,t,3)
+ (localhost,57640,100806,t,3)
+ (localhost,57640,100807,t,3)
+(8 rows)
+
+-- we shouldn't have any prepared transactions in the workers
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             0
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100804,t,3)
+ (localhost,57637,100805,t,3)
+ (localhost,57637,100806,t,3)
+ (localhost,57637,100807,t,3)
+ (localhost,57640,100804,t,3)
+ (localhost,57640,100805,t,3)
+ (localhost,57640,100806,t,3)
+ (localhost,57640,100807,t,3)
+(8 rows)
+
+-- kill as soon as the coordinator sends COMMIT
+SELECT citus.mitmproxy('conn.onQuery(query="^COMMIT PREPARED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table DROP COLUMN new_column;
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+-- some of the placements would be missing the new column
+-- since we've not commited the prepared transactions
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     2
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100804,t,2)
+ (localhost,57637,100805,t,2)
+ (localhost,57637,100806,t,2)
+ (localhost,57637,100807,t,2)
+ (localhost,57640,100804,t,3)
+ (localhost,57640,100805,t,3)
+ (localhost,57640,100806,t,3)
+ (localhost,57640,100807,t,3)
+(8 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is committed
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             4
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100804,t,2)
+ (localhost,57637,100805,t,2)
+ (localhost,57637,100806,t,2)
+ (localhost,57637,100807,t,2)
+ (localhost,57640,100804,t,2)
+ (localhost,57640,100805,t,2)
+ (localhost,57640,100806,t,2)
+ (localhost,57640,100807,t,2)
+(8 rows)
+
+-- finally, test failing on ROLLBACK with 2CPC
+-- fail just after the coordinator sends the ROLLBACK
+-- so the command can be rollbacked
+SELECT citus.mitmproxy('conn.onQuery(query="ROLLBACK").kill()');
```

Comment:
I still think this is not a particularly useful test. We're suppressing WARNING so we're not really resting any of the coordinator's behavior. And by killing connections all we're exercising is the Postgres behavior: automatically rolling back when the backend ends.

Reply:
Okay, good points.